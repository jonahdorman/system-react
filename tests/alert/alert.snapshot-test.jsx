/* eslint-env jest */
import { testDOMandHTML } from '../snapshot-helpers';

import Info from '../../examples/alert/info';
import Warning from '../../examples/alert/warning';
import ErrorAlert from '../../examples/alert/error';
import Offline from '../../examples/alert/offline';
import Dismissable from '../../examples/alert/dismissable';
import CustomClassNames from '../../examples/alert/custom-class-name';

testDOMandHTML({
	name: 'Alert Info',
	test,
	Component: Info
});

testDOMandHTML({
	name: 'Alert Warning',
	test,
	Component: Warning
});

testDOMandHTML({
	name: 'Alert Error',
	test,
	Component: ErrorAlert
});

testDOMandHTML({
	name: 'Alert Offline',
	test,
	Component: Offline
});

testDOMandHTML({
	name: 'Alert Dismissable',
	test,
	Component: Dismissable
});

testDOMandHTML({
	name: 'Alert Custom Class Name',
	test,
	Component: CustomClassNames
});
