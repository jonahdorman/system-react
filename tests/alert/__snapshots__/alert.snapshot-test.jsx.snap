// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Custom Class Name DOM Snapshot 1`] = `
<IconSettings
  iconPath="/assets/icons"
>
  <SLDSAlertContainer
    className="this-is-the-container"
  >
    <SLDSAlert
      assistiveText={
        Object {
          "closeButton": "Close",
        }
      }
      className="this-is-the-alert"
      icon={
        <SLDSIcon
          category="utility"
          name="user"
          size="medium"
        />
      }
      labels={
        Object {
          "heading": "Logged in as John Smith (johnsmith@acme.com).",
          "headingLink": "Log out",
        }
      }
      onClickHeadingLink={[Function]}
      variant="info"
    />
  </SLDSAlertContainer>
</IconSettings>
`;

exports[`Alert Custom Class Name HTML Snapshot 1`] = `
"<div class=\\"slds-notify-container this-is-the-container\\">
  <div class=\\"slds-notify slds-notify_alert slds-theme_alert-texture slds-theme_info this-is-the-alert\\" role=\\"alert\\"><span class=\\"slds-assistive-text\\">info</span><span class=\\"slds-m-right--x-small\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon slds-icon--x-small\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#user\\"></use></svg></span>
    <h2>Logged in as John Smith (johnsmith@acme.com). <a href=\\"javascript:void(0);\\">Log out</a></h2>
  </div>
</div>"
`;

exports[`Alert Dismissable DOM Snapshot 1`] = `
<IconSettings
  iconPath="/assets/icons"
>
  <SLDSAlertContainer>
    <SLDSAlert
      assistiveText={
        Object {
          "closeButton": "Close",
        }
      }
      dismissible={true}
      icon={
        <SLDSIcon
          category="utility"
          name="user"
          size="medium"
        />
      }
      labels={
        Object {
          "heading": "Logged in as John Smith (johnsmith@acme.com).",
          "headingLink": "Log out",
        }
      }
      onClickHeadingLink={[Function]}
      onRequestClose={[Function]}
      variant="info"
    />
  </SLDSAlertContainer>
</IconSettings>
`;

exports[`Alert Dismissable HTML Snapshot 1`] = `
"<div class=\\"slds-notify-container\\">
  <div class=\\"slds-notify slds-notify_alert slds-theme_alert-texture slds-theme_info\\" role=\\"alert\\"><span class=\\"slds-assistive-text\\">info</span><span class=\\"slds-m-right--x-small\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon slds-icon--x-small\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#user\\"></use></svg></span>
    <h2>Logged in as John Smith (johnsmith@acme.com). <a href=\\"javascript:void(0);\\">Log out</a></h2><button class=\\"slds-button slds-button--icon-inverse slds-notify__close\\" title=\\"Close\\" type=\\"button\\"><svg aria-hidden=\\"true\\" class=\\"slds-button__icon\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#close\\"></use></svg><span class=\\"slds-assistive-text\\">Close</span></button></div>
</div>"
`;

exports[`Alert Error DOM Snapshot 1`] = `
<IconSettings
  iconPath="/assets/icons"
>
  <SLDSAlertContainer>
    <SLDSAlert
      assistiveText={
        Object {
          "closeButton": "Close",
        }
      }
      icon={
        <SLDSIcon
          category="utility"
          name="ban"
          size="medium"
        />
      }
      labels={
        Object {
          "heading": "Your browser is currently not supported. Your Salesforce may be degraded.",
          "headingLink": "More Information",
        }
      }
      onClickHeadingLink={[Function]}
      variant="error"
    />
  </SLDSAlertContainer>
</IconSettings>
`;

exports[`Alert Error HTML Snapshot 1`] = `
"<div class=\\"slds-notify-container\\">
  <div class=\\"slds-notify slds-notify_alert slds-theme_alert-texture slds-theme_error\\" role=\\"alert\\"><span class=\\"slds-assistive-text\\">error</span><span class=\\"slds-m-right--x-small\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon slds-icon--x-small\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#ban\\"></use></svg></span>
    <h2>Your browser is currently not supported. Your Salesforce may be degraded. <a href=\\"javascript:void(0);\\">More Information</a></h2>
  </div>
</div>"
`;

exports[`Alert Info DOM Snapshot 1`] = `
<IconSettings
  iconPath="/assets/icons"
>
  <SLDSAlertContainer>
    <SLDSAlert
      assistiveText={
        Object {
          "closeButton": "Close",
        }
      }
      icon={
        <SLDSIcon
          category="utility"
          name="user"
          size="medium"
        />
      }
      labels={
        Object {
          "heading": "Logged in as John Smith (johnsmith@acme.com).",
          "headingLink": "Log out",
        }
      }
      onClickHeadingLink={[Function]}
      variant="info"
    />
  </SLDSAlertContainer>
</IconSettings>
`;

exports[`Alert Info HTML Snapshot 1`] = `
"<div class=\\"slds-notify-container\\">
  <div class=\\"slds-notify slds-notify_alert slds-theme_alert-texture slds-theme_info\\" role=\\"alert\\"><span class=\\"slds-assistive-text\\">info</span><span class=\\"slds-m-right--x-small\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon slds-icon--x-small\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#user\\"></use></svg></span>
    <h2>Logged in as John Smith (johnsmith@acme.com). <a href=\\"javascript:void(0);\\">Log out</a></h2>
  </div>
</div>"
`;

exports[`Alert Offline DOM Snapshot 1`] = `
<IconSettings
  iconPath="/assets/icons"
>
  <SLDSAlertContainer>
    <SLDSAlert
      assistiveText={
        Object {
          "closeButton": "Close",
        }
      }
      labels={
        Object {
          "heading": "You are in offline mode.",
          "headingLink": "More information",
        }
      }
      onClickHeadingLink={[Function]}
      variant="offline"
    />
  </SLDSAlertContainer>
</IconSettings>
`;

exports[`Alert Offline HTML Snapshot 1`] = `
"<div class=\\"slds-notify-container\\">
  <div class=\\"slds-notify slds-notify_alert slds-theme_alert-texture slds-theme_offline\\" role=\\"alert\\"><span class=\\"slds-assistive-text\\">offline</span><span class=\\"slds-m-right--x-small\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon slds-icon--x-small\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#offline\\"></use></svg></span>
    <h2>You are in offline mode. <a href=\\"javascript:void(0);\\">More information</a></h2>
  </div>
</div>"
`;

exports[`Alert Warning DOM Snapshot 1`] = `
<IconSettings
  iconPath="/assets/icons"
>
  <SLDSAlertContainer>
    <SLDSAlert
      assistiveText={
        Object {
          "closeButton": "Close",
        }
      }
      labels={
        Object {
          "heading": "Your browser is outdated. Your Salesforce experience may be degraded.",
          "headingLink": "More Information",
        }
      }
      onClickHeadingLink={[Function]}
      variant="warning"
    />
  </SLDSAlertContainer>
</IconSettings>
`;

exports[`Alert Warning HTML Snapshot 1`] = `
"<div class=\\"slds-notify-container\\">
  <div class=\\"slds-notify slds-notify_alert slds-theme_alert-texture slds-theme_warning\\" role=\\"alert\\"><span class=\\"slds-assistive-text\\">warning</span><span class=\\"slds-m-right--x-small\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon slds-icon--x-small\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#warning\\"></use></svg></span>
    <h2>Your browser is outdated. Your Salesforce experience may be degraded. <a href=\\"javascript:void(0);\\">More Information</a></h2>
  </div>
</div>"
`;
